### PR TITLE
Added deserialization by rowGroupIndex

### PR DESCRIPTION
**Pull request by mcbos**
…on https://github.com/aloneguid/parquet-dotnet/issues/329

### CODE REVIEW

File: docs/serialisation.md [hunk 1]
```diff
@@ -38,11 +38,19 @@ That's it! Of course the `.SerializeAsync()` method also has overloads and optio
...
 
 ```csharp
 IList<Record> data = await ParquetSerializer.DeserializeAsync<Record>("/mnt/storage/data.parquet");
 ```
+## Deserialize records by RowGroup
+
+If you have a large file and you want to deserialize it in chunks, you can also read records by row group. This can help to keep memory usage low as you won't need to load the entire file into memory.
+
+```csharp
+IList<Record> data = await ParquetSerializer.DeserializeByRowGroupAsync<Record>("/mnt/storage/data.parquet", rowGroupIndex);
```
Comment (aloneguid) [inline docs/serialisation.md:51]: `DeserializeByRowGroupAsync` method doesn't seem to exist.

File: docs/serialisation.md [hunk 2]
```diff
@@ -4,11 +4,11 @@ Parquet library is generally extremely flexible in terms of supporting internals
 
 Class serialisation is **really fast** as internally it generates [compiled expression trees](https://learn.microsoft.com/en-US/dotnet/csharp/programming-guide/concepts/expression-trees/) on the fly. That means there is a tiny bit of delay when serialising a first entity, which in most cases is negligible. Once the class is serialised at least once, further operations become blazingly fast (around *x40* speed improvement comparing to reflection on relatively large amounts of data (~5 million records)).
 
-Class serialisation philosophy is based on the idea that we don't need to reinvent the wheel when it comes to converting objects to and from JSON. Instead of creating our own custom serialisers and deserialisers, we can leverage the existing JSON infrastructure that .NET provides. This way, we can save time and effort, and also make our code more consistent and compatible with other .NET applications that use JSON.
```
Comment (aloneguid) [inline docs/serialisation.md:7]: It's `deserialise` in British English, so no big deal ;) Methods are called with `z` just to look similar to other methods in .net.

File: src/Parquet/Serialization/ParquetSerializer.cs [hunk 3]
```diff
@@ -102,53 +102,102 @@ public static class ParquetSerializer {
...
         /// </summary>
         /// <typeparam name="T"></typeparam>
         /// <param name="source"></param>
+        /// <param name="rowGroupIndex"></param>
         /// <param name="options"></param>
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
         /// <exception cref="InvalidOperationException"></exception>
         public static async Task<IList<T>> DeserializeAsync<T>(Stream source,
+            int rowGroupIndex,
```
Comment (aloneguid) [inline src/Parquet/Serialization/ParquetSerializer.cs:117]: might be worth adding after `options` parameter and give it a default value, that way it's minimum breaking changes.

File: src/Parquet/Serialization/ParquetSerializer.cs [hunk 4]
```diff
@@ -102,53 +102,102 @@ public static class ParquetSerializer {
...
+            using ParquetReader reader = await ParquetReader.CreateAsync(source, options, cancellationToken: cancellationToken);
+            for(int rgi = 0; rgi < reader.RowGroupCount; rgi++) {
+                await DeserializeRowGroupAsync(reader, rgi, asm, result, cancellationToken);
+            }
+
+            return result;
+        }
+
+        private static Assembler<T> GetAssembler<T>() where T : new()
+        {
```
Comment (aloneguid) [inline src/Parquet/Serialization/ParquetSerializer.cs:159]: Formatting seems to be broken in a few places. I'm not sure your editor/IDE supports `.editorconfig` which is configured with preferred rules for this project.

**Review by aloneguid — CHANGES_REQUESTED** [review]

File: src/Parquet/Serialization/ParquetSerializer.cs (hunk 3 above)
Comment (mcbos) [reply src/Parquet/Serialization/ParquetSerializer.cs:117]: I'm using method overloading to not break current functionality.

See [line 140](https://github.com/mcbos/parquet-dotnet/blob/deserialize-by-rowgroup/src/Parquet/Serialization/ParquetSerializer.cs#L140).

File: src/Parquet/Serialization/ParquetSerializer.cs (hunk 4 above)
Comment (mcbos) [reply src/Parquet/Serialization/ParquetSerializer.cs]: Rider was doing strange things, hope its better now.

**Review by aloneguid — APPROVED** [review]
lgtm, thanks!

**Pull request merged**